### DISCUSSION TITLE
Pocketbase Cascading

### DISCUSSION THREAD

**Discussion by Vrajs16** [pocketbase/pocketbase#1996] [Q&A]
I remember before upgrading to pocketbase 13.2, I was able to do cascading delete for a relation.

For example lets say that I have a collection called products and each record has a relation to multiple records in an collection called images. 

Before I remember I was able to delete one record in the products collection and all the related images in the images collection would be deleted. 

As of now that is not happening?

**Comment by ganigeorgiev** [comment]
There are no changes to the records cascade handling in the latest couple releases. We also have integrations and unit tests that verify the behavior.

I've also just tested it on the live demo (https://pocketbase.io/demo) and it works as expected for me.

> Before I remember I was able to delete one record in the products collection and all the related images in the images collection would be deleted.

I'm not sure that I understand the above described structure, but the cascade operation is triggered on relation record delete -> delete the ref record, aka. in your case the `images` record(s) will be deleted on `products` delete if the `images` has a cascade `relation` field pointing to the `products`.

If that's not what you are experiencing, please share your collections structure (you can export it from Admin UI > Settings > Export Collections).

**Comment by Vrajs16** [comment]
Ok that is strange I thought it working but it isn't. Sorry about that. Then this brings me to my next question.

How would I structure this db? 

Because if I call the js sdk to give me all the products and I expand the relation for image column, I would get the images that I want for that product record. However lets say that product is no longer needed and has to be deleted. Wouldn't it be better to delete the product record and all sub relations should be deleted instead of the deleting the images and that would delete the product record?

Here is my collections structure:
```json
[
    {
        "id": "95exa8gtayejmq3",
        "name": "images",
        "type": "base",
        "system": false,
        "schema": [
            {
                "id": "kboh43tc",
                "name": "image",
                "type": "url",
                "system": false,
                "required": false,
                "unique": false,
                "options": {
                    "exceptDomains": null,
                    "onlyDomains": null
                }
            }
        ],
        "listRule": null,
        "viewRule": null,
        "createRule": null,
        "updateRule": null,
        "deleteRule": null,
        "options": {}
    },
    {
        "id": "pw9uz8lh74hcpdr",
        "name": "products",
        "type": "base",
        "system": false,
        "schema": [
            {
                "id": "d2v3awbj",
                "name": "field",
                "type": "relation",
                "system": false,
                "required": false,
                "unique": false,
                "options": {
                    "collectionId": "95exa8gtayejmq3",
                    "cascadeDelete": true,
                    "maxSelect": 1,
                    "displayFields": []
                }
            }
        ],
        "listRule": null,
        "viewRule": null,
        "createRule": null,
        "updateRule": null,
        "deleteRule": null,
        "options": {}
    }
]

Thanks!

**Reply by ganigeorgiev** [reply]
I'm not familiar with your use case and I'm not sure what to suggest or recommend.

If you want on `products` record delete to delete automatically all related `images`, **you have to inverse the relation**, aka. define the `relation` field in the `images` collection. The `relation` field could be single (in case an image can be associate to only 1 product) or multiple (in case a single image can be used by multiple products).

Alternatively, you can also fallback to a more traditional SQL approach using "junction" collection (eg. `productImageRel`) that could have 2 `relation` fields for the `products` and the `images` collections.

But again, this depends on your use case and it is difficult to recommend a single solution without the full picture.
